File: atlhyper_master_v2/model/observe_logs.go
// atlhyper_master_v2/model/observe_logs.go
// Observe Logs 查询参数与结果类型
package model

import "AtlHyper/model_v3/log"

// LogSnapshotQueryOpts 快照日志查询选项
type LogSnapshotQueryOpts struct {
	Service   string
	Level     string
	Scope     string
	StartTime string
	EndTime   string
	Offset    int
	Limit     int
}

// LogSnapshotResult 快照日志查询结果
type LogSnapshotResult struct {
	Logs   []log.Entry
	Total  int
	Facets log.Facets
}

File: atlhyper_web/src/api/observe-logs.ts
/**
 * Logs 信号域 API
 */

import { get, post } from "./request";
import type { ObserveResponse } from "./observe-common";
import type { LogEntry, LogFacets, LogHistogramResult } from "@/types/model/log";

/** 日志查询结果（不含 histogram） */
export interface LogQueryResponse {
  logs: LogEntry[];
  total: number;
  facets: LogFacets;
}

/** 查询日志 (POST) */
export function queryLogs(params: {
  cluster_id: string;
  query?: string;
  service?: string;
  level?: string;
  scope?: string;
  trace_id?: string;
  span_id?: string;
  limit?: number;
  offset?: number;
  since?: string;
  start_time?: string;
  end_time?: string;
}) {
  return post<ObserveResponse<LogQueryResponse>>("/api/v2/observe/logs/query", params);
}

/** 查询日志直方图 (GET, ClickHouse 聚合) */
export function getLogsHistogram(params: {
  cluster_id: string;
  since?: string;
  service?: string;
  level?: string;
  scope?: string;
  query?: string;
  start_time?: string;
  end_time?: string;
}) {
  return get<ObserveResponse<LogHistogramResult>>("/api/v2/observe/logs/histogram", params);
}
